alert/views.py

```diff
@@ -233,3 +236,14 @@ def send_json(style, msg):
         return do_register(course_code, email_address, phone, SOURCE_API, send_json, key)
     else:
         raise Http404('Only POST requests are accepted.')
+
+
+class RegistrationViewSet(viewsets.ModelViewSet):
+    serializer_class = RegistrationSerializer
+    http_method_names = ['get', 'post', 'put']
+    permission_classes = [IsAuthenticated]
+
+    def get_queryset(self):
+        queryset = Registration.objects.filter(user=self.request.user)
+        queryset = super().get_serializer_class().setup_eager_loading(queryset)
```

Comment:
Remove `setup_eager_loading()` since we now have AutoPrefetch, but if we want PCR annotations, we can do the `prefetch_related` directly in `get_queryset()`.